src/sqlfluff/core/plugin/host.py

```diff
@@ -25,6 +29,56 @@
 is_main_process: ContextVar[bool] = ContextVar("is_main_process", default=True)
 
 
+def _get_sqlfluff_version() -> str:
+    """Get the SQLFluff package version from importlib.
+
+    NOTE: At the stage of loading plugins, SQLFluff isn't fully
+    initialised and so we can't use the normal methods.
+    """
+    return importlib.metadata.version("sqlfluff")
+
+
+def _discover_plugins() -> Iterator[Tuple[importlib.metadata.EntryPoint, str, str]]:
+    """Uses the same mechanism as pluggy to introspect available plugins.
+
+    This method is then intended to allow loading of plugins individually,
+    for better error handling.
+    """
+    for dist in list(importlib.metadata.distributions()):
+        for ep in dist.entry_points:
+            # Check it's a SQLFluff one
+            if ep.group != project_name:
+                continue
+            yield ep, ep.name, dist.version
+
+
+def _load_plugin(
+    plugin_manager: pluggy.PluginManager,
+    entry_point: importlib.metadata.EntryPoint,
+    plugin_name: str,
+    plugin_version: str,
+) -> None:
+    """Loads a single plugin with a bit of error handling."""
+    # NOTE: If the plugin is already loaded, then .register() will fail,
+    # so it's important that we check whether it's loaded at this point.
+    if plugin_manager.get_plugin(plugin_name):  # pragma: no cover
+        plugin_logger.info("...already loaded")
+        return None
+    try:
+        plugin = entry_point.load()
+    except Exception as err:
+        plugin_logger.error(
```

Comment:
This will log the error message but no stack backtrace. Could a stack backtrace be useful here?